Automattic/simplenote-ios
# Tags List Refresh: Mark II

### Details:
In this PR we're (massively) refactoring **SPTagListViewController**:

- Everything is now a row. No more buttons within the tableViewCell
- New **SPTagHeaderView** class
- **SPTagListCell** is now Autolayout friendly (and hence, dynamic type)
- **SPTagListViewController**'s initialization is properly split into few methods
- **SPTagListViewController** no longer performs manual layout calculation

@aerych @danielebogo @SergioEstevao Gentlemen... I'm sorry about flooding you with code, so much, so often!!!

**Thank you!!!**

Ref. #408

### Testing: New UI // cc @aerych (Thank you sir!!)
1. Log into your account
2. Press over the top left button to reveal the Tags List

- [x] Verify that the `Settings` button shows up below `Trash`
- [x] Verify that Dark Mode looks as expected
- [x] Verify that Dynamic Type gets everything to properly resize
- [x] Verify that the Tags List properly remembers the previously selected cell (close and reopen)

### Testing: Tag Edition via Edition Button  // cc @danielebogo  (Thank you Daniele!!!)
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Press over the top left button to reveal the Tags List
5. Press the `Edit` button

- [x] Verify that **only** tag rows can be moved
- [x] Verify that **only** tag rows can be deleted
- [x] Verify that **only** tag rows can be renamed
- [x] Verify that while a Tag row is being renamed, the tableView's bottom insets are properly adjusted: you should be able to scroll down to the bottom.

### Testing: Tag Edition via MenuController // cc @SergioEstevao (Thank you Sergio!!)
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Perform a long press gesture over any of the tags

- [ ] Verify that the highlighted tag can be renamed
- [ ] Verify that the highlighted tag can be deleted
- [ ] Verify that a long press over `All Notes` / `Trash` / `Settings` do not trigger such menu

### Testing: Tags Sort Order // cc @SergioEstevao  (Thank you Sergio!!)
1. Log into your account
2. Edit any random note and add several tags
3. Open the Notes List
4. Press over the top left button to reveal the Tags List
5. Press over `Settings`
6. Change the `Tags` sort order

- [ ] Verify that the Tags List gets properly updated

---

### Screenshot!

<img src="https://user-images.githubusercontent.com/1195260/66855097-ff3f2680-ef58-11e9-8007-a87bc8113034.png" width="200">
